manual/src/cmds/tail-mod-cons.etex

```diff
@@ -150,63 +152,6 @@ tail-mod-cons are as follows:
   using a convenient monadic notation.
 \end{itemize}
 
-\section{sec:disambiguation}{Disambiguation}
-
-It may happen that several arguments of a constructor are recursive
-calls to a tail-modulo-cons function. The transformation can only turn
-one of these calls into a tail call. The compiler will not make an
-implicit choice, but ask the user to provide an explicit
-disambiguation.
-
-Consider this type of syntactic expressions (assuming some
-pre-existing type "var" of expression variables):
-\begin{caml_example*}{verbatim}
-type var (* some pre-existing type of variables *)
-
-type exp =
-  | Var of var
-  | Let of binding * exp
-and binding = var * exp
-\end{caml_example*}
-
-Consider a "map" function on variables. The direct definition has two
-recursive calls inside arguments of the "Let" constructor, so it gets
-rejected as ambiguous.
-\begin{caml_example*}{verbatim}[error]
-let[@tail_mod_cons] rec map_vars f exp =
-  match exp with
-  | Var v -> Var (f v)
-  | Let ((v, def), body) ->
-    Let ((f v, map_vars f def), map_vars f body)
-\end{caml_example*}
-
-To disambiguate, the user should add a "[\@tailcall]" attribute to the
-recursive call that should be transformed to tail position:
-\begin{caml_example*}{verbatim}
-let[@tail_mod_cons] rec map_vars f exp =
-  match exp with
-  | Var v -> Var (f v)
-  | Let ((v, def), body) ->
-    Let ((f v, map_vars f def), (map_vars[@tailcall]) f body)
-\end{caml_example*}
-Be aware that the resulting function is \emph{not} tail-recursive, the
-recursive call on "def" will consume stack space. However, expression
-trees tend to be right-leaning (lots of "Let" in sequence,
-rather than nested inside each other), so putting the call on "body"
-in tail position is an interesting improvement over the naive
-definition: it gives bounded stack space consumption if we assume
-a bound on the nesting depth of "Let" constructs.
-
-One would also get an error when using conflicting annotations, asking
-for two of the constructor arguments to be put in tail position:
-\begin{caml_example*}{verbatim}[error]
-let[@tail_mod_cons] rec map_vars f exp =
-  match exp with
-  | Var v -> Var (f v)
-  | Let ((v, def), body) ->
-    Let ((f v, (map_vars[@tailcall]) f def), (map_vars[@tailcall]) f body)
-\end{caml_example*}
```

Comment:
I would expect that the manual keeps an example of conflicting annotations: It seems reasonable to explain that only one argument of a constructor can be transformed into a tail-call?